src/Node.js

```diff
@@ -0,0 +1,121 @@
+  // the properties that a regular DOM node would contain, but only the ones
+  // needed by slowparse.
+module.exports = (function(){
+  "use strict";
+
+  var voidHtmlElements = require("./voidHtmlElements");
+
+  function Node(nodeType, nodeName, nodeValue, namespaceURI) {
+    this.nodeType = nodeType;
+    this._nodeName = nodeName;
+    this.nodeValue = nodeValue || "";
+    this.namespaceURI = namespaceURI || "";
+    this.parentNode = false;
+    this.nextSibling = false;
+    this.isVoid = voidHtmlElements.indexOf(nodeName) > -1;
+    this.childNodes = [];
+    this.attributes = [];
+    this._attributeMap = {};
+  }
+
+  Node.prototype = {
+    // Add a node to the current node as a child
+    appendChild: function(node) {
+      var lastChild = this.childNodes.slice(-1)[0];
+      if(lastChild) {
+        lastChild.nextSibling = node;
+      }
+
+      node.parentNode = this;
+      this.childNodes.push(node);
+    },
+    // Create a deep copy of the current node
+    cloneNode: function() {
+      var node = new Node(this.nodeType, this.nodeName, this.nodeValue, this.namespaceURI);
+      node.nextSibling = this.nextSibling ? this.nextSibling.cloneNode() : false;
+      node.isVoid = this.isVoid;
+      node.childNodes = this.childNodes.map(function(childNode) {
+        return childNode.cloneNode();
+      });
+      node.attributes = this.attributes.map(function(attribute) {
+        return attribute.cloneNode();
+      });
+      node._attributeMap = this._attributeMap;
+
+      return node;
+    },
+    // Get the attribute value of an element node
+    getAttribute: function(name) {
+      return this.nodeType === Node.ELEMENT_NODE && this._attributeMap[name.toLowerCase()];
+    }
+  };
+
+  Object.defineProperties(Node.prototype, {
+    "nodeName": {
+      get: function() {
+        if(this.nodeType === Node.ELEMENT_NODE) {
+          return this._nodeName.toUpperCase();
+        }
+        if(this.nodeType === Node.ATTRIBUTE_NODE) {
+          return this._nodeName.toLowerCase();
+        }
+
+        return this._nodeName;
+      }
+    },
+    "innerHTML": {
+      get: function() {
+        if(this.nodeType !== Node.ELEMENT_NODE) {
+          return null;
+        }
+
+        var innerHTML = "";
+
+        this.childNodes.forEach(function(childNode) {
+          innerHTML += childNode.outerHTML || "";
+        });
+
+        return innerHTML;
+      }
+    },
+    "outerHTML": {
+      get: function() {
+        if(this.nodeType === Node.ATTRIBUTE_NODE) {
+          return null;
+        }
+        if(this.nodeType === Node.COMMENT_NODE) {
+          return "<!--" + this.nodeValue + "-->";
+        }
+        if(this.nodeType !== Node.ELEMENT_NODE) {
+          return this.nodeValue;
+        }
+
+        var attributes = this._attributeMap;
+        var attributeStr = "";
+        attributeStr = Object.keys(attributes).map(function(attrName) {
+          return attrName + "=\"" + attributes[attrName] + "\"";
```

Comment:
We have two quotation symbols, they make things easier to read: `return attrName + '="' + attributes[attrName] + '"';`